repository-s3/src/main/java/io/aiven/elasticsearch/repositories/s3/S3RepositoryStorageIOProvider.java

```diff
@@ -0,0 +1,271 @@
+            }
+        }
+
+        @Override
+        public Tuple<Integer, Long> deleteDirectories(final String path) throws IOException {
+            final List<Tuple<String, Long>> listOfFiles;
+            try {
+                //we need only prefix here since it is impossible to distinguish between "files" and "directories"
+                //in AWS SDK e.g. paths like:
+                // - /aaaa/bbb/ccc/file
+                // - /aaa/bbb/file
+                // with request which contains delimiter and without returns diff result, and ES API doesn't use
+                // recursive calls to get full list of files by path/sub-path
+                listOfFiles = Permissions.doPrivileged(() ->
+                        list(new ListObjectsV2Request().withBucketName(bucketName).withPrefix(path)))
+                        .stream()
+                        .flatMap(l -> l.getObjectSummaries().stream())
+                        .map(o -> Tuple.tuple(o.getKey(), o.getSize()))
+                        .collect(Collectors.toList());
+            } catch (final AmazonClientException e) {
+                throw new IOException("Couldn't get list of files for path " + path, e);
+            }
+
+            deleteFiles(listOfFiles.stream()
+                            .map(Tuple::v1)
+                            .collect(Collectors.toList()),
+                    true);
+
+            final var removedBytes = listOfFiles.stream().map(Tuple::v2).reduce(0L, Long::sum);
+            return Tuple.tuple(listOfFiles.size(), removedBytes);
+        }
+
+        @Override
+        public void deleteFiles(final List<String> blobNames,
+                                final boolean ignoreIfNotExists) throws IOException {
+
+            final var partitionList = new ArrayList<List<String>>() {
+
+                static final int MAX_DELETE_BULK = 10_000;
+
+                @Override
+                public List<String> get(final int index) {
+                    final var start = index * MAX_DELETE_BULK;
+                    final var end = Math.min(start + MAX_DELETE_BULK, blobNames.size());
+                    return blobNames.subList(start, end);
+                }
+
+                @Override
+                public int size() {
+                    return (int) Math.ceil((double) blobNames.size() / (double) MAX_DELETE_BULK);
+                }
+            };
+
+            for (var idx = 0; idx < partitionList.size(); idx++) {
+                final var chunk = partitionList.get(idx);
+                final var deleteObjectsRequest =
+                        new DeleteObjectsRequest(bucketName)
+                                .withKeys(chunk.toArray(new String[0]))
+                                .withQuiet(true);
+                try {
+                    Permissions.doPrivileged(() -> client.deleteObjects(deleteObjectsRequest));
+                } catch (final MultiObjectDeleteException e) {
```

Comment:
`withQuiet` and `client.deleteObjects`'s result ignored together results in errors ignored.

Reply:
It doesn't meter for S3 use withQuite or not ... it will skip unknown objects in the bucket ...